satosa_oidcop/idpy_oidcop.py

```diff
@@ -350,10 +350,17 @@ def token_endpoint(self, context: ExtendedContext):
             return self.send_response(proc_req)
         elif isinstance(proc_req, TokenErrorResponse):
             return self.send_response(JsonResponse(proc_req.to_dict(), status="403"))
-
+        
+        # TODO: remove when migrate to idpy-oidc
+        # PATCH https://github.com/UniversitaDellaCalabria/SATOSA-oidcop/issues/29
+        if isinstance (proc_req["response_args"].get('scope', str), list):
+            proc_req["response_args"]['scope'] =  ' '.join(proc_req["response_args"]['scope'])
```

Comment:
This is the crux of the change and it very much makes sense and aligns with the spec [1]: convert a list to space-separated string.

Just a bit of nit-picking: is there a reason for mixing the double and single quotes in a single expression?

[1] https://www.rfc-editor.org/rfc/rfc6749#section-3.3

Reply:
I should run Blake on this!
Thank you, I'll align the quotes to be always twice

Reply:
Thanks, just saw a lot of changes from the lint.

Just wondering: what line-width is the linting set to?

Some of the code looked more readable to me on a single line than split.

It might be just a matter of preference - but making sure we're not formatting to "historical terminal screens".

Reply:
linting here
https://github.com/UniversitaDellaCalabria/SATOSA-oidcop/blob/token_endpoint_res/linting.sh

Reply:
Thanks - 120 chars is reasonable.